dallinger/recruiters.py

```diff
@@ -287,7 +287,7 @@ def validate_config(self, **kwargs):
         """Validates config variables. Override this method for recruiter-specific validation."""
         if not self.supports_delayed_publishing:
             assert self.config.get(
-                "publish_experiment"
+                "publish_experiment", True
```

Comment:
@pmcharrison If unset  (MTurk) we pass the assertion

Reply:
Perfect!

Reply:
Use publish_experiment_default here?

Reply:
Sorry, I'm being stupid today.... thank you for working this all out!!

Reply:
Yes, we can merge now.

Reply:
no, wait, let me think some more

Reply:
It is a tricky thing now with 3 variables included all within one condition. And to me _to allow deployment if either (a) publish_experiment = False_ isn't what I thought it should be.

Reply:
> Not sure why you think so. Don't we want to make sure that people accidentally think they can make a draft deployment when they can't as in the case of MTurk?

So we don't care about the case of an intended delayed deployment?

Reply:
Sorry, you're right. I should have said 

> we want to allow deployment if either (a) publish_experiment = True in config, or (b) if publish_experiment is missing from config, because in the latter case we will default to publish_experiment_default, which is True for MTurk.

Reply:
So what you want is
```
assert self.config.get(
        "publish_experiment", self.publish_experiment_default
    ), f"{type(self).__name__} does not support delayed experiment publishing. Set `publish_experiment=true` in your experiment config!"
``` ?

Reply:
Following my logic above, where I proposed `publish_experiment_default = not supports_delayed_publishing`, then that means that when we are in this if statement (`if not supports_delayed_publishing`) then we know that `publish_experiment_default = True`, so we might as well replace `True` with `publish_experiment_default` to enhance readability. 

To put it another way, we are just writing a more advanced version of `config.get(` that takes into account the default value.

Reply:
OK. Pushed. Last check?

Reply:
> So what you want is
> 
> ```
> assert self.config.get(
>         "publish_experiment", self.publish_experiment_default
>     ), f"{type(self).__name__} does not support delayed experiment publishing. Set `publish_experiment=true` in your experiment config!"
> ```
> 
> ?


Yep!